app/src/androidTest/java/org/sagebionetworks/research/mpower/RoomScheduledActivityTests.kt

```diff
@@ -0,0 +1,239 @@
+package org.sagebionetworks.research.mpower
+
+import android.arch.persistence.room.Room
+import android.support.test.InstrumentationRegistry
+import android.support.test.runner.AndroidJUnit4
+import junit.framework.Assert.assertEquals
+import junit.framework.Assert.assertNotNull
+import junit.framework.Assert.assertNull
+import junit.framework.Assert.assertTrue
+import org.joda.time.DateTime
+
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.sagebionetworks.bridge.rest.RestUtils
+import org.sagebionetworks.bridge.rest.model.ScheduleStatus
+import org.sagebionetworks.bridge.rest.model.ScheduledActivityListV4
+import org.sagebionetworks.research.mpower.room.ResearchDatabase
+import org.sagebionetworks.research.mpower.room.RoomScheduledActivity
+import org.sagebionetworks.research.mpower.room.RoomScheduledActivityDao
+import org.sagebionetworks.research.mpower.room.RoomTypeConverters
+import java.io.IOException
+import java.nio.charset.Charset
+
+@RunWith(AndroidJUnit4::class)
+class RoomScheduledActivityTests {
+
+    private var database: ResearchDatabase? = null
+    private var activityDao: RoomScheduledActivityDao? = null
+    private var activityList: ScheduledActivityListV4? = null
+    private var roomActivityList: List<RoomScheduledActivity>? = null
+
+    @Before @Test
+    fun setupRoom() {
```

Comment:
Is this the equivalent of Before or BeforeClass? Do we have any concerns about cross-contamination between the individual tests in this class? If not, it might be more efficient to use the equivalent of BeforeClass and setup and teardown the in memory DB once for the whole class.

Reply:
I'm a little embarrassed, but I had no idea that @Before meant before each test, I thought it was equivalent to @BeforeClass.  I'll fix this up.

Reply:
Nice, tests run about 3x as fast now.